dd-trace-core/src/main/java/datadog/trace/core/PendingTrace.java

```diff
@@ -448,4 +448,8 @@ public static long getDurationNano(CoreSpan<?> span) {
     PendingTrace trace = (PendingTrace) traceCollector;
     return trace.getLastWriteTime() - span.getStartTime();
   }
+
+  public ConcurrentLinkedDeque<DDSpan> getSpans() {
```

Comment:
Avoid exposing the implementation - since the caller only iterates over the result this could be:
```suggestion
  public Iterable<DDSpan> getSpans() {
```